### ISSUE TITLE
[Bug] : preset doesn't allow enable plugins

### ISSUE THREAD

**Issue by alexander-akait** [cssnano/cssnano#1098]
**Describe the bug**

preset doesn't allow to enable plugins, only disable

**To Reproduce**
https://github.com/webpack-contrib/css-minimizer-webpack-plugin/issues/87

or just set:

```
preset: [
  'lite',
  {
    discardDuplicates: true,
  },
],
```

CSS:

```
h1 {
    margin: 0 auto;
    margin: 0 auto
}

h1 {
    margin: 0 auto
}
```

Output

```css
h1{margin:0 auto;margin:0 auto}h1{margin:0 auto}
```

**Expected behavior**

Plugin should enabled and minify to `h1{margin:0 auto}`

**Screenshots**

No need

**Desktop (please complete the following information):**
- CSSNANO Version [e.g. 22] latest
- Plugins/presets versions lite
- Add the output of `npx envinfo && npm ls cssnano` here
---

**Additional context**

/cc @ludofischer I think critical for us

**Comment by ludofischer** [comment]
I think the reason is that `preset-lite` does not include `discard-duplicates` at all. We would need to restructure the presets so every preset depends on all plugins. I suppose the main disadvantage would be that the install size of the smaller presets would increase.

**Comment by alexander-akait** [comment]
We can add peerDeps for yarn PnP, ask to install plugin and when you have `discard-duplicates: true` require it

**Comment by ludofischer** [comment]
If you want to be able to enable any plugin from a single package, we could fold the different plugins into a single package. Imagine having a single `OnceExit` where at every step (declaration, etc). we apply every relevant transformation in order. I think it would make the code easier to read and to maintain and probably faster  (accordinng to a benchmark I've seen it's twice as slow as the other minimizers). It would not necessarily increase the package size a lot (currently cssnano ships a 400kb YAML parser just because it uses cosmiconfig which supports YAML by default)

**Comment by alexander-akait** [comment]
@ludofischer Yes, once upon a time it was my idea for perf improvements, idea was - remove presets and refactor postcss plugins to functions and use them inside `cssnano`, maybe we can do it now too without breaking change (under option(s))

**Comment by ludofischer** [comment]
> @ludofischer Yes, once upon a time it was my idea for perf improvements, idea was - remove presets and refactor postcss plugins to functions and use them inside `cssnano`, maybe we can do it now too without breaking change (under option(s))

Yes!  To see if we can avoid any breaking changes I would need to understand how the current configuration works in practice, because I think it does not work as the current docs say. There are a few bugs open (https://github.com/cssnano/cssnano/issues/1055, https://github.com/cssnano/cssnano/issues/904), but maybe it might be better to keep it working as it does now, 'fixing` to do as the docs say it a this point would be a breaking change.

We could move the transformations to a separate package (say 'cssnano core') which has an interface for activating the different transforms. cssnano would still call separate plugins (there are third-party ones anyway) but more and more transforms would go into `cssnano-core`. In the end,  the cssnano package would turn into just a configuration reader. The current plugins would still exist for some time but switch to just call  'cssnano-core' with a single transform turned on (we could deprecate them once they all use 'cssnano-core').

**Comment by alexander-akait** [comment]
Sounds good :+1:
